frontend/src/page/places.vue

```diff
@@ -635,7 +635,14 @@ export default {
         }
       });
 
-      this.map.on('idle', this.updateMarkers);
+      // Following example for cluster updates from here: https://maplibre.org/maplibre-gl-js/docs/examples/cluster-html/
+      this.map.on('data', (e) => {
+          if (e.sourceId !== 'photos' || !e.isSourceLoaded) return;
+
+          this.map.on('move', this.updateMarkers);
```

Comment:
Wouldn't this attach the event handlers every time the results are updated, for example when the user performs a search? In this case, I would expect the map to slow down with each search performed.

Reply:
Related Documentation: https://maplibre.org/maplibre-gl-js/docs/API/types/maplibregl.MapDataEvent/

Reads like that could happen. The example they provide might be a special case, in case the data is loaded once. If you don't have time to test it, I can do it later today.